osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ConfigurationService.java

```diff
@@ -108,134 +111,89 @@ public ConfigurationService() {
   }
 
   public void setSpecialDays(
-      final MessageMetadata deviceMessageMetadata,
-      final org.opensmartgridplatform.domain.core.valueobjects.smartmetering.SpecialDaysRequest
-          specialDaysRequestValueObject)
+      final MessageMetadata messageMetadata, final SpecialDaysRequest specialDaysRequest)
       throws FunctionalException {
 
     log.info(
         "setSpecialDays for organisationIdentification: {} for deviceIdentification: {}",
-        deviceMessageMetadata.getOrganisationIdentification(),
-        deviceMessageMetadata.getDeviceIdentification());
+        messageMetadata.getOrganisationIdentification(),
+        messageMetadata.getDeviceIdentification());
 
-    final SmartMeter smartMeteringDevice =
-        this.domainHelperService.findSmartMeter(deviceMessageMetadata.getDeviceIdentification());
+    final SmartMeter smartMeter =
+        this.domainHelperService.findSmartMeter(messageMetadata.getDeviceIdentification());
 
-    final SpecialDaysRequestDto specialDaysRequestDto =
-        this.configurationMapper.map(specialDaysRequestValueObject, SpecialDaysRequestDto.class);
+    final SpecialDaysRequestDto requestDto =
+        this.configurationMapper.map(specialDaysRequest, SpecialDaysRequestDto.class);
 
     this.osgpCoreRequestMessageSender.send(
-        new RequestMessage(
-            deviceMessageMetadata.getCorrelationUid(),
-            deviceMessageMetadata.getOrganisationIdentification(),
-            deviceMessageMetadata.getDeviceIdentification(),
-            smartMeteringDevice.getIpAddress(),
-            specialDaysRequestDto),
-        deviceMessageMetadata.getMessageType(),
-        deviceMessageMetadata.getMessagePriority(),
-        deviceMessageMetadata.getScheduleTime(),
-        deviceMessageMetadata.isBypassRetry());
+        requestDto, messageMetadata.builder().withIpAddress(smartMeter.getIpAddress()).build());
   }
 
   public void setConfigurationObject(
-      final MessageMetadata deviceMessageMetadata,
-      final org.opensmartgridplatform.domain.core.valueobjects.smartmetering
-              .SetConfigurationObjectRequest
-          setConfigurationObjectRequestValueObject)
+      final MessageMetadata messageMetadata,
+      final SetConfigurationObjectRequest setConfigurationObjectRequest)
       throws FunctionalException {
 
     log.info(
         "setConfigurationObject for organisationIdentification: {} for deviceIdentification: {}",
-        deviceMessageMetadata.getOrganisationIdentification(),
-        deviceMessageMetadata.getDeviceIdentification());
+        messageMetadata.getOrganisationIdentification(),
+        messageMetadata.getDeviceIdentification());
 
-    final SmartMeter smartMeteringDevice =
-        this.domainHelperService.findSmartMeter(deviceMessageMetadata.getDeviceIdentification());
+    final SmartMeter smartMeter =
```

Comment:
We discussed this: not for now...

Reply:
Possibly an opportunity to refactor further. Two things stopped me: one is that at some point you stop refactoring, or you run the risk of going on too long, the other is that these are mostly one liners, where I could imagine that for instance with the builder there could reappear some variation again, and it would have to be fit in the refactored code. Could be I am mistaken in my assessment of what could happen, and the refactoring is still a good idea, but then my first argument still is there.